### DISCUSSION TITLE
Declaring marks used by the defining package and other packages

### DISCUSSION THREAD

**Discussion by jwodder** [pytest-dev/pytest#9913] [Q&A]
We have a package (call it `cheese`) that contains (in addition to the primary code) various test-helper functions & utility decorators that, among other things, apply custom markers to decorated tests.  Various other packages that depend on `cheese` make use of these helpers in their own tests, but because the markers are (currently) only defined in the `tox.ini` file in `cheese`'s repository, warnings about unknown markers are emitted when testing these other packages.  The best way to solve this appears to be to declare the markers in a `pytest_configure` hook, but that leads to the following questions:

1. Is the only way to make the `pytest_configure` hook take effect for other packages to turn `cheese` into a pytest plugin?

2. Where should we put the `pytest_configure` hook so that it takes effect for `cheese`'s own tests?  Note that `cheese`'s `conftest.py` file contains some autouse fixtures which we do not want used by other packages, so I expect that putting the hook in that file and pointing the pytest plugin to it would cause problems.

**Comment by yarikoptic** [comment]
ping @pytest-dev team on this question
